jogo42-cb/main.c

```diff
@@ -18,6 +18,16 @@ void MoverJog(GameState* gs);
 // Move os obstaculos
 void MoverObst(GameState* gs);
 
+const float dis=100.0;
+
+bool ataque (GameState* gs)
+{
+    Vector2 Atk;
+    Atk.x= gs->jog.pos.x + dis*cos(gs->jog.rot/60);
```

Comment:
- Tá faltando espaços.
- As funções `sin` e `cos` do `math.h` usam números do tipo `double` tanto como argumentos quanto como valor de retorno, mas os números que nós usamos no jogo são todos `float` (ex: `Jog.pos`, `Jog.rot`, `Atk`, `dis`, etc). 

  Esse código funciona porque ele faz conversão implícita entre `float` e `double`. Mas isso não é ideal, porque é mais lento e pode resultar em imprecisões. O ideal seria usar as funções `sinf` e `cosf` também do `math.h`, que fazem a mesma coisa porém com `float`s.
```suggestion
    Atk.x = gs->jog.pos.x + dis * cosf(gs->jog.rot / 60);
```